ocf_datapipes/select/select_pv_systems_without_output_wholeday.py

```diff
@@ -0,0 +1,63 @@
+"""Select PV systems and Dates with No output for the entire day"""
+import logging
+from collections import defaultdict
+from datetime import datetime
+from typing import Dict, List, Union
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+logger = logging.getLogger(__name__)
+
+
+@functional_datapipe("select_pv_systems_without_output")
+class SelectPVSystemsWithoutOutputIterDataPipe(IterDataPipe):
+    """Function to select pv system ids with dates that has no output whatsoever in a given day.
+
+    Returns a two key pair (pvsystem id and date) dictionary with values as the status
+    "Active" or "Inactive" in a given day.
+
+    Yield example:
+        default dict(<class dict>,{"10003:{"2020-04-01 : "Inactive", "2020-05-01" : "Active"}, "10004": {"2020-04-01" : "Active"}})
+
+        Args:
+            source_datapipe: Datapipe of Xarray Dataset emitting timeseries data
+    """
+
+    def __init__(self, source_datapipe: IterDataPipe) -> None:
+
+        self.source_datapipe = source_datapipe
+
+    def __iter__(self) -> xr.Dataset():
+
+        for xr_dataset in self.source_datapipe:
+            xr_dataset = self.source_datapipe
+            dates_list = xr_dataset.coords["datetime"].values
+            ssid_list = list(xr_dataset)
+            dates_list = [
+                datetime.strptime(str(x)[:10], "%Y-%m-%d").strftime("%Y-%m-%d") for x in dates_list
```

Comment:
you can use `.data` is simplier

Reply:
I would also go with using the times directly, rather than converting them and using them as dates again later.